integration/fuse/src/main/java/alluxio/jnifuse/AbstractFuseFileSystem.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.jnifuse;
+
+import alluxio.jnifuse.struct.FileStat;
+import alluxio.jnifuse.struct.FuseFileInfo;
+import alluxio.jnifuse.struct.Statvfs;
+import alluxio.jnifuse.utils.SecurityUtils;
+import alluxio.util.OSUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Abstract class for other File System to extend and integrate with Fuse.
+ */
+public abstract class AbstractFuseFileSystem implements FuseFileSystem {
+
+  static {
+    System.loadLibrary("jnifuse");
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractFuseFileSystem.class);
+
+  private static final int TIMEOUT = 2000; // ms
```

Comment:
Should this be configurable somehow?

Reply:
this is only used for initialization, not worthy.
I added more javadoc